components/brave_new_tab_ui/components/default/rewards/index.tsx

```diff
@@ -51,7 +50,7 @@ export interface RewardsProps {
 }
 
 function getVisibleGrant (promotions: NewTab.Promotion[]): GrantInfo | null {
-  if (promotions.length === 0) {
+  if (!promotions || promotions.length === 0) {
```

Comment:
good check but if we're nor marking `promotions` as optional here, should we be checking if the array is `undefined` higher up in the stack? Like for whetaver renders the `<RewardsWidget` and passes the `promotions` prop in?

Reply:
Good call - that change doesn't make a whole lot of sense from a TS point of view. I've updated the PR to make the component prop optional instead.

I would like to see us modify `new_tab_storage.ts:load` such that we maintain type invariants when we deserialize data from local storage, but I can create a follow-up for that.